src/zip.ts

```diff
@@ -0,0 +1,119 @@
+import fs from 'fs';
+import path from 'path';
+import { execSync } from 'child_process';
+import { platform } from 'os';
+import { errCode } from './error';
+
+function validate(workspace: string, source: string, dest: string) {
+  if (!source.trim()) {
+    throw new Error('The source path cannot be empty.');
+  }
+  if (!dest.trim()) {
+    throw new Error('The destination file cannot be empty.');
+  }
+
+  // Verify the source folder exists and is not a file.
+  try {
+    const stats = fs.statSync(source);
+    if (!stats.isDirectory()) {
+      throw new Error('Invalid source folder: the source must be a directory.');
+    }
+  } catch (err) {
+    if (errCode(err) === 'ENOENT') {
+      throw new Error('Invalid source folder: not exist.');
+    }
+    throw new Error(`Failed to access source folder: ${err}`);
```

Comment:
(optional) Do we need this distinction? Or is the `ENOENT` error case's message not good enough that we need our own? Happy to keep it as is, just wanted to know for my own knowledge 🙏

Reply:
I added it just for any unknown error here. But I'm also fine to remove it. From my experiments, I can't get the error from this line 😄